cheddar/cheddar-integration-aws/src/main/java/com/clicktravel/infrastructure/messaging/aws/sns/SnsTopicResource.java

```diff
@@ -49,7 +53,12 @@ public SnsTopicResource(final String topicName, final String topicArn, final Ama
      * @throws AmazonClientException
      */
     public void publish(final String subject, final String message) throws AmazonClientException {
-        amazonSnsClient.publish(new PublishRequest().withTopicArn(topicArn).withSubject(subject).withMessage(message));
+        final Map<String, MessageAttributeValue> attributes = new HashMap<String, MessageAttributeValue>();
```

Comment:
This is fine, but please also add an inline comment giving the reason why this message attribute is being added i.e. mention it is intended for use with the SNS Message Filtering feature.